frontend_tests/node_tests/util.js

```diff
@@ -169,3 +169,59 @@ var _ = global._;
         assert(!util.is_all_or_everyone_mentioned(messages_without_everyone_mentions[i]));
     }
 }());
+
+(function test_move_array_elements_to_front() {
+    var strings = [
+        'string1',
+        'string3',
+        'string2',
+        'string4',
+    ];
+    var strings_selection = [
+        'string4',
+        'string1',
+    ];
+    var strings_expected = [
+        'string1',
+        'string4',
+        'string3',
+        'string2',
+    ];
+    var strings_no_selection = util.move_array_elements_to_front(strings, []);
+    var strings_no_array = util.move_array_elements_to_front([], strings_selection);
+    var strings_actual = util.move_array_elements_to_front(strings, strings_selection);
+    var emails = [
+        'test@zulip.com',
+        'test@test.com',
+        'test@localhost',
+        'test@invalid@email',
+        'something@zulip.com',
+    ];
+    var emails_selection = [
+        'test@test.com',
+        'test@localhost',
+        'test@invalid@email',
+    ];
+    var emails_expected = [
+        'test@test.com',
+        'test@localhost',
+        'test@invalid@email',
+        'test@zulip.com',
+        'something@zulip.com',
+    ];
+    var emails_actual = util.move_array_elements_to_front(emails, emails_selection);
+    var i;
+    assert(strings_no_selection.length === strings.length);
+    for (i=0; i<strings_no_selection.length; i += 1) {
```

Comment:
just formatting issue:
`i=0` and `i += 1` spaces or lack of spaces around? ;-)

Please choose a good one (based on codebase) and provide the formatting in all of your code.